tinygrad/runtime/ops_ios.py

```diff
@@ -0,0 +1,239 @@
+    
+    for (j,i),input_idx in self.input_replace.items():
+      computeCommand = self.device.msg(self.icb, "indirectComputeCommandAtIndex:", j, res=new_var())
+      self.device.msg(computeCommand, "setKernelBuffer:offset:atIndex:", input_rawbuffers[input_idx]._buf.buf,
+                                                                                 input_rawbuffers[input_idx]._buf.offset, i)
+
+    for j, global_dims, local_dims in self.updated_launch_dims(var_vals):
+      prg = cast(CompiledRunner, self.jit_cache[j].prg)
+      global_size, local_size = global_dims or prg.p.global_size, local_dims or prg.p.local_size
+      computeCommand = self.device.msg(self.icb, "indirectComputeCommandAtIndex:", j,res=new_var())
+      self.device.msg(computeCommand,"concurrentDispatchThreadgroups:threadsPerThreadgroup:",global_size[0],global_size[1],global_size[2],local_size[0],local_size[1],local_size[2])
+    
+    for j, var in enumerate(self.vars):
+      self.int_buf_view[j] = var_vals[var]
+
+    if len(self.vars) > 0:
+      formatted_hex = ' '.join(f'{b:02x}' for b in self.int_buf_view.tobytes())
+      self.device.msg("copyin",formatted_hex,self.int_buf.buf)
+
+    command_buffer = self.device.msg(self.device.mtl_queue,"commandBuffer",res=new_var())
+    encoder = self.device.msg(command_buffer,"computeCommandEncoder",res=new_var())
+    self.device.msg(encoder,"useResources:count:usage:",*all_resources,
+            "MTLResourceUsage.MTLResourceUsageRead | MTLResourceUsage.MTLResourceUsageWrite")
+
+    self.device.msg(encoder,"executeCommandsInBuffer:withRange:",self.icb,len(self.jit_cache))
+    self.device.msg(encoder,"endEncoding")
+    self.device.msg(command_buffer,"commit")
+    self.command_buffer = command_buffer
+
+    self.device.mtl_buffers_in_flight.append(command_buffer)
+    if wait:
+      for cbuf in self.device.mtl_buffers_in_flight: self.device.msg(cbuf,"waitUntilCompleted")
+      self.device.mtl_buffers_in_flight.clear()
+      return float(self.device.msg(command_buffer,"elapsed_time"))
+    return None
+
+class IOSDevice(Compiled):
+  def __init__(self, device:str):
+    self.device = "d"
+    self.ip = "http://192.168.1.6:8081" #iphone ip address
+    self.queue = []
+    self.files = set()
+    #self.msg("delete","x")
+    self.mtl_queue = self.msg(self.device,"newCommandQueueWithMaxCommandBufferCount:",1024,res=new_var())
+    self.mtl_buffers_in_flight: List[Any] = []
+
+    super().__init__(device, IOSAllocator(self), iOSRenderer(), Compiler(),
+                     functools.partial(IOSProgram, self), IOSGraph)
+    
+
+  def msg(self,ptr,selector,*args,res=None):
+    req = [ptr,selector]
+    req.append(len(args))
+    for x in args: req.append(x)
+    if res != None: req.append(res)
+    self.queue.append(req)
+    if selector in ["copyout","maxTotalThreadsPerThreadgroup","elapsed_time","file_exists"] or len(self.queue) > 1000: #todo, don't send bytes as a string etc
```

Comment:
todo?

Reply:
I had a size limit of 1000 for the queues because the iOS app couldn't handle large requests, I've fixed that now